server/src/routes/flora/users.ts

```diff
@@ -18,43 +20,51 @@ router.get("/", async (_: Request, res: Response) => {
 router.post("/login", async (req: Request, res: Response) => {
   const uusername: string = req.body.uusername;
   const upassword: string = req.body.upassword;
-  if (uusername === undefined) {
-    return res
-      .status(HttpStatusCode.BAD_REQUEST)
-      .json({ error: "No username field provided" });
-  } else if (upassword === undefined) {
-    return res
-      .status(HttpStatusCode.BAD_REQUEST)
-      .json({ error: "No password field provided" });
+
+  if (!uusername || !upassword) {
+    throw new ApiError(
+      HttpStatusCode.BAD_REQUEST,
+      ApiErrorMessage.FloraLogin.MISSING_FIELDS
+    );
   }
 
   const query =
     "SELECT uid, ucredits FROM Users where " +
     `uusername='${uusername}' and upassword='${upassword}'`;
   const { rows } = await getDb().query(query);
-  if (rows.length === 0) {
-    return res
-      .status(HttpStatusCode.NOT_FOUND)
-      .json({ error: "User not found" });
+  const queriedUser: Pick<User, "uid" | "ucredits"> | undefined = rows[0];
+
+  if (!queriedUser) {
+    throw new ApiError(
+      HttpStatusCode.NOT_FOUND,
+      ApiErrorMessage.FloraLogin.USER_NOT_FOUND
+    );
   }
 
-  return jwt.sign(rows[0], "secretkey", (err: any, token: any) => {
-    return res.status(HttpStatusCode.OK).json({ token });
-  });
+  const user: User = { ...queriedUser, uusername, upassword };
+  const token: string = jwt.sign(user, JWT_SECRET_KEY);
+
+  return res.status(HttpStatusCode.OK).json({ token, user });
 });
 
-// GET request - get user inventory
-router.get("/inventory", verifyToken, async (req: Request, res: Response) => {
-  if (req.body.user === undefined) {
-    return res
-      .status(HttpStatusCode.BAD_REQUEST)
-      .json({ error: "No uid field provided" });
+// GET request - get a user's inventory
+router.get("/:id/inventory", verifyJwt, async (req: Request, res: Response) => {
+  const user: User = res.locals.user;
```

Comment:
@Yonggiee to get the `User` from the JWT token, call `res.locals.user`, which is guaranteed to be well defined if `verifyJwt` does not throw any errors. If `verifyJwt` throws any errors, it will automatically be handled by `customErrorHandler` and show the callee appropriate messages.